docs/source/markdown/podman-login.1.md

```diff
@@ -28,6 +28,12 @@ For more details about format and configurations of the auth.json file, please r
 
 ## OPTIONS
 
+#### **--accept-repositories**
```

Comment:
I still question why is this an option?  Why would I care? I don't like to add options that know one is ever going to choose?  Should this option default to true?  If so who would ever set it to false?

Reply:
I think my question may mirror@rhatdan's.  Do we need this option at all?  If a user specifies `podman login quay.io/username/image` could we just log into that rather than `quay.io.`?

Reply:
Follow-up of https://github.com/containers/common/pull/659#discussion_r673842324, https://github.com/containers/common/pull/659#issuecomment-872999764

@vrothberg @mtrmac would it make sense to automatically switch on the flag if a repository is provided?

Reply:
In theory it is a breaking change if we did.  If @rhatdan wants to take the risk, I am OK with it.

Reply:
```
 podman login quay.io/rhatdan/podman
Username: rhatdan
Password: 
Login Succeeded!
```
Currently works.  But this is undefined behavior, I believe.

With the new change will push quay.io/rhatdan/buildah fail?

Reply:
Right now it will only write `quay.io` into auth.json, where with the new behavior (and the flag enabled) it will write `quay.io/rhatdan/podman`.

Reply:
Meaning the podman push of quay.io/rhatdan/buildah will fail because the repo is not in the quay.io/rhatdan/podman.

Currently the login to quay.io/rhatdan/podman writes to quay.io/rhatdan, correct?

Reply:
> Meaning the podman push of quay.io/rhatdan/buildah will fail because the repo is not in the quay.io/rhatdan/podman.

No, it will fallback to the credentials previously written, aka `quay.io` .

Ref: https://github.com/containers/image/blob/9e1d7b48f1de2ea1ffe61a74916f6b0bcd605abb/pkg/docker/config/config.go#L677-L697

> Currently the login to quay.io/rhatdan/podman writes to quay.io/rhatdan, correct?

Currently, a login to `quay.io/rhatdan/podman` writes only `quay.io`.

After this merge, everything works as before without breaking existing behavior if the flag is not being used.

Reply:
(Stripping repo from the argument)
> But this is undefined behavior, I believe.

Well, it’s an explicitly made change , in https://github.com/containers/podman/pull/1905 , for https://github.com/containers/podman/issues/1690 .

---

We have added the opt-in flag exactly because this is a breaking change. It would make no sense to _both_ have the flag and have it default to on; the impacted users would have to add the flag to revert to the previous behavior, so they might just as well change the non-option argument.

(It might make sense to force the flag to on, without having it in the CLI at all. That would be equally a breaking change (and something c/common does not currently allow, I think), but at least the users wouldn’t be bothered with the existence of the flag.

Overall I don’t think the breaking change is the best option — Podman does have user reports of users using the repo syntax, and expecting it to work in a way compatible with Docker. So the opt-in not to break users makes more sense to me. But it’s a judgement call; I could see the argument that many users passing a repo without the opt-in flag are likely to be making a mistake with the old semantics.)